### ISSUE TITLE
Comments are not handled correctly

### ISSUE THREAD

**Issue by slarse** [KTH/spork#35]
Or... at all? I'm not sure what I'm even doing about comments. But it's certainly not being handled correctly.

**Comment by slarse** [comment]
I'm considering just dropping comments for the time being.

**Comment by monperrus** [comment]
dropping at resolution time or at pretty-printing time?

**Comment by slarse** [comment]
Resolution time, I think that comments should either be handled correctly all the way through, or not at all. Support for comments can then be added later, if time permits. It can also be done incrementally, for example supporting type comments (e.g. class, interface, etc) is quite simple.

It's still more work, though, and I'm feeling some slight time pressure now.

**Comment by slarse** [comment]
However, with the new architecture I'm currently adding in to handle modifiers #36 and type parameter bounds #37, adding support for comments may actually not be much work at all. I'll have a definitive answer by the end of the week.

**Comment by slarse** [comment]
@monperrus I've essentially spent the entire day dealing with merging comments. I actually do think they are really important, and several other structured or semi-structured tools that I've tried simply drop inline comments, and don't properly merge the raw text of comments. Realistically, a merge tool that does not respect comments and documentation is not useful in practice.

So now I'm textually merging comments, and am able to output comments in most cases. It's again a fight with the pretty printer, as it uses the positions of comments to determine where to output them. And since I'm merging trees from three different sources, the positions of comments in relation to a node don't necessarily make any sense.

I've worked around this by unsetting the positions entirely. Comments may sometimes pop from below a node to above it, but they should not disappear. I hope.
